File: backend/src/main/java/com/ll/hotel/domain/member/member/type/BusinessApprovalStatus.java
package com.ll.hotel.domain.member.member.type;

public enum BusinessApprovalStatus {
    PENDING("승인 대기"),
    APPROVED("승인 완료"),
    REJECTED ("승인 거절"),
    CANCELED("승인 취소");

    private String description;

    BusinessApprovalStatus(String description) {
        this.description = description;
    }
}

File: frontend/src/lib/types/business/BusinessApprovalStatus.ts
export enum BusinessApprovalStatus {
  PENDING = "PENDING",
  APPROVED = "APPROVED",
  REJECTED = "REJECTED",
}

// 코드 → 한글 변환
export const BusinessApprovalStatusLabels: Record<
  BusinessApprovalStatus,
  string
> = {
  [BusinessApprovalStatus.PENDING]: "승인 대기",
  [BusinessApprovalStatus.APPROVED]: "승인 완료",
  [BusinessApprovalStatus.REJECTED]: "승인 거절",
};

// 한글 → 코드 변환
export const ReverseBusinessApprovalStatusMap = Object.fromEntries(
  Object.entries(BusinessApprovalStatusLabels).map(([key, value]) => [
    value,
    key,
  ])
) as Record<string, BusinessApprovalStatus>;
